### PR TITLE
Update olm addon to v0.17.0

### LINKED ISSUE #10768: update the Outdated catalog in OLM addon
I was wondering why my minikube instance is installing old operator versions.

Then I found out that the `CatalogSource` image in the OLM addon points to the SHA of the image that is several months old.  https://github.com/kubernetes/minikube/blob/81eeb7dd979e29a712ac5153fc4b6b40bd519f9f/pkg/minikube/assets/addons.go#L294

It might make sense for this one case to be an exception from https://github.com/kubernetes/minikube/issues/9884 and to use  `latest` tag. Otherwise, every user will have to update the image refernce themself to pick up the latest changes in Operator Catalog.

### PR DESCRIPTION
**Pull request by kadel**
- update olm addon to use OLM v0.17.0

fixes #10768

### CODE REVIEW

File: deploy/addons/olm/olm.yaml.tmpl [hunk 1]
```diff
@@ -78,9 +78,9 @@ spec:
           command:
           - /bin/olm
           args:
-          - -namespace
+          - --namespace
```
Comment (medyagh) [inline deploy/addons/olm/olm.yaml.tmpl:81]: what is the reason for the additional -- ?

**Review by medyagh — CHANGES_REQUESTED** [review]
the crds.yaml.tmpl diff says 
```
11,302 additions, 10,357 deletions not shown because the diff is too large. Please use a local Git client to view these changes.
```
this doesnt sound right, do u mind clarifying what is inside that file?

**Comment by kadel** [comment]
> the crds.yaml.tmpl diff says
> 
> ```
> 11,302 additions, 10,357 deletions not shown because the diff is too large. Please use a local Git client to view these changes.
> ```
> 
> this doesnt sound right, do u mind clarifying what is inside that file?

those are CustomeResourceDefinitions used by olm taken from v0.17.0 release. You can check it at https://github.com/operator-framework/operator-lifecycle-manager/releases/tag/v0.17.0

File: deploy/addons/olm/olm.yaml.tmpl (hunk 1 above)
Comment (kadel) [reply deploy/addons/olm/olm.yaml.tmpl:81]: old OLM versions were using `flags` at some point they switched to `github.com/spf13/pflag` library, hence move from `-` to `--`

**Comment by sharifelgamal** [comment]
Hey @kadel, we have a test (in test/integration/addons_test.go) that currently skip testing the OLM addon. Could you remove the skip so that we can properly test this PR?

Thanks!

**Comment by kadel** [comment]
> Hey @kadel, we have a test (in test/integration/addons_test.go) that currently skips testing the OLM addon. Could you remove the skip so that we can properly test this PR?
> 
> Thanks!

oh, I haven't noticed the integration test. 
Done.

File: deploy/addons/olm/olm.yaml.tmpl [hunk 2]
```diff
@@ -326,8 +328,16 @@ spec:
                   requests:
                     cpu: 10m
                     memory: 50Mi
+                securityContext:
+                  runAsUser: 1000
```
Comment (medyagh) [inline deploy/addons/olm/olm.yaml.tmpl:332]: is this required ? this seems to give Root Access to the addon,
do u mind clarifying why we need to run with more special access ?

File: deploy/addons/olm/olm.yaml.tmpl [hunk 3]
```diff
@@ -308,7 +310,7 @@ spec:
                 - --global-namespace
                 - olm
                 image: {{.CustomRegistries.OLM  | default .ImageRepository | default .Registries.OLM }}{{.Images.OLM}}
-                imagePullPolicy: IfNotPresent
+                imagePullPolicy: Always
```
Comment (medyagh) [inline deploy/addons/olm/olm.yaml.tmpl:313]: is this change related to bumping the version ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: deploy/addons/olm/olm.yaml.tmpl (hunk 2 above)
Comment (kadel) [reply deploy/addons/olm/olm.yaml.tmpl:332]: I don't work on OLM directly I'm just a user and I did not write this definition. This comes from the official OLM release artifacts. 

But If I'm not mistaken root has uid `0`. This is not giving any special access to the container, it is doing quite the opposite. It ensures that the container can't run anything as root. 
`1000` is the first non-root user.  Even if the image is built in a way that it runs as root inside the container, this will force it to be non-root.

File: deploy/addons/olm/olm.yaml.tmpl (hunk 3 above)
Comment (kadel) [reply deploy/addons/olm/olm.yaml.tmpl:313]: yes

File: deploy/addons/olm/olm.yaml.tmpl (hunk 2 above)
Comment (medyagh) [reply deploy/addons/olm/olm.yaml.tmpl:332]: that makes sense ! thanks for the clarifying

**Comment by medyagh** [comment]
Looks good to me @kadel  , Thank you for this PR and improving olm addon

look forward for more PRs for you
